src/dftbp/solvation/gbsafile.F90

```diff
@@ -216,62 +219,26 @@ subroutine nextLine(unit, line, lineno, file, iostat, node)
     !> Error code
     integer, intent(out), optional :: iostat
 
-    integer :: length, iErr
+    integer :: iErr
     character(len=lc) :: errorStr, iomsg
 
     lineno = lineno + 1
+    call lineReader%readLine(line, iErr)
 
-    read(unit, '(a)', advance='no', iostat=iErr, iomsg=iomsg, size=length) line
-    if (length >= len(line)) then
+    if (present(iostat)) then
+      iostat = iErr
+    else if (iErr /= 0) then
       if (present(file)) then
```

Comment:
Am I missing something, or is file always present in calls to this routine?

Reply:
Fair enough. I've reduced the routine now to the bare minimum according to the actual use cases.